File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int res = -1, dist = Integer.MAX_VALUE;
        for (int i = 0; i < points.length; i++) {
            int xx = points[i][0], yy = points[i][1];
            if (xx == x || yy == y) {
                int tmp = Math.abs(xx - x) + Math.abs(yy - y);
                if (tmp < dist) {
                    dist = tmp;
                    res = i;
                }
            }
        }
        return res;
    }
}

File: 1779. Find Nearest Point That Has the Same X or Y Coordinate/Solution.py
import math
from typing import List


class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        res, dist = -1, math.inf
        for i, (xx, yy) in enumerate(points):
            if xx == x or yy == y:
                tmp = abs(xx - x) + abs(yy - y)
                if tmp < dist:
                    dist = tmp
                    res = i
        return res
